statina/API/v2/endpoints/batches.py

```diff
@@ -0,0 +1,288 @@
+    return JSONResponse(
+        content=jsonable_encoder(
+            dict(
+                sex_thresholds={
+                    "XY_fetal_fraction_y": sex_thresholds.XY_fetal_fraction_y(),
+                    "XX_lower": sex_thresholds.XX_lower(),
+                    "XX_upper": sex_thresholds.XX_upper(),
+                    "XY_upper": sex_thresholds.XY_upper(),
+                    "XY_lower": sex_thresholds.XY_lower(),
+                    "XXY": sex_thresholds.XXY(),
+                },
+                control=control,
+                abnormal=abnormal_dict,
+                cases=cases,
+                max_x=x_max,
+                min_x=x_min,
+            )
+        ),
+    )
+
+
+@router.get("/batch/{batch_id}/fetal_fraction")
+def fetal_fraction(
+    batch_id: str,
+    current_user: User = Security(get_current_active_user, scopes=["R"]),
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+):
+    """Batch view with fetal fraction plot"""
+    return JSONResponse(
+        content=jsonable_encoder(
+            dict(
+                control=get_fetal_fraction.samples(
+                    adapter=adapter, batch_id=batch_id, control_samples=True
+                ),
+                cases=get_fetal_fraction.samples(adapter=adapter, batch_id=batch_id),
+            )
+        ),
+    )
+
+
+@router.get("/batch/{batch_id}/coverage")
+def coverage(
+    batch_id: str,
+    current_user: User = Security(get_current_active_user, scopes=["R"]),
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+):
+    """Batch view with coverage plot"""
+    db_samples: List[DataBaseSample] = find.batch_samples(batch_id=batch_id, adapter=adapter)
+    samples: List[Sample] = [Sample(**db_sample.dict()) for db_sample in db_samples]
+
+    scatter_data: Dict[str, CoveragePlotSampleData] = get_scatter_data_for_coverage_plot(samples)
+    box_data: Dict[int, List[float]] = get_box_data_for_coverage_plot(samples)
+    return JSONResponse(
+        content=jsonable_encoder(
+            dict(
+                x_axis=list(range(1, 23)),
+                scatter_data=scatter_data,
+                box_data=box_data,
+            )
+        ),
+    )
+
+
+@router.put("/batch/{batch_id}/comment")
+async def batch_update_comment(
+    batch_id: str,
+    comment: Optional[str] = Form(...),
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+    current_user: User = Security(get_current_active_user, scopes=["RW"]),
+):
+    """Update batch comment"""
+
+    batch: Batch = find.batch(batch_id=batch_id, adapter=adapter)
+    batch.comment = comment
+    update.update_batch(adapter=adapter, batch=batch)
+
+    return JSONResponse(content="Batch comment updated", status_code=202)
+
+
+@router.patch("/batch/{batch_id}/include_samples")
+async def include_samples(
+    batch_id: str,
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+    current_user: User = Security(get_current_active_user, scopes=["RW"]),
```

Comment:
So including just some samples is solved by having a separate include sample endpoint..? I like.

Reply:
No need for save button \__anywhere__